src/Algorithms/CARTTrainer.cpp

```diff
@@ -310,37 +310,36 @@ CARTTrainer::TreeType CARTTrainer::buildTree(AttributeTables const& tables, Clas
 			//Continue recursively
 			nodeInfo.attributeIndex = bestAttributeIndex;
 			nodeInfo.attributeValue = bestAttributeVal;
-			
+
 
 			//Store entry in the tree
 			nodeInfo.leftNodeId = nodeId+1;
 			TreeType lTree = buildTree(lTables, dataset, cBestBelow, nodeInfo.leftNodeId);
 			nodeInfo.rightNodeId = nodeInfo.leftNodeId+ lTree.size();
 			TreeType rTree = buildTree(rTables, dataset, cBestAbove, nodeInfo.rightNodeId);
-			
+
 			TreeType tree;
 			tree.push_back(nodeInfo);
 			tree.insert(tree.end(), lTree.begin(), lTree.end());
 			tree.insert(tree.end(), rTree.begin(), rTree.end());
 			return tree;
 		}
 	}
-	
+
 	TreeType tree;
 	tree.push_back(nodeInfo);
 	return tree;
 }
 
-RealVector CARTTrainer::hist(boost::unordered_map<std::size_t, std::size_t> countMatrix){
+RealVector CARTTrainer::hist(ClassVector const& countVector) const {
 
```

Comment:
Wow, thank you, I had completely forgotten about that! Initially `countVector` was a `std::vector`, I changed it to a UIntVector specifically for this reason, but then just completely forgotten about it. I'll put the impurity functions through the same magic :D